buildSrc/src/main/kotlin/packetevents.library-conventions.gradle.kts

```
@@ -0,0 +1,73 @@
+plugins {
+    `java-library`
+    `maven-publish`
+}
+
+group = rootProject.group
+version = rootProject.version
+description = rootProject.description
+
+repositories {
+    mavenLocal()
+    mavenCentral()
+    maven("https://oss.sonatype.org/content/groups/public/")
+}
+
+dependencies {
+    compileOnly("org.jetbrains:annotations:23.0.0")
+}
+
+java {
+    withSourcesJar()
+    disableAutoTargetJvm()
+}
```

Perhaps, as @SamB440 said earlier in this PR, we should set the toolchain to use Java 21. Due to your compiler settings, this will still compile back to Java 8.

```suggestion
java {
    toolchain.languageVersion = JavaLanguageVersion.of(21)
    withSourcesJar()
    disableAutoTargetJvm()
}
```

---

We should be forcing the language version of the code to be Java 8. Hence, this shouldn't be done. As of @SamB440, he is introducing the sponge module which would require java version to be 21. This shouldn't be set globally as our own code should remain Java 8. And for the fabric, its already set to Java 17.

---

Thanks for the clarification! About Fabric specifically, that should probably be updated from Java 17 -> 21.

---

Is it necessary and is Fabric backwards compatible?

---

Not that backwards compatibility is essential on any platform other than Spigot, but if it is not necessary why bother now?

---

This pr is not for updating fabric, I shouldn't merge two key update in one pr.

---

I agree with you Abhigya. Fabric isn't relevant for this PR. Anyway, the PR is really nice! It's much easier to navigate through the build chain now.

---

👍